### PR TITLE
[test] Add test for Fieldextra COSMO postprocessing

### PR DESCRIPTION
**Pull request by kraushm**
Fieldextra tests (cookbook + performance test) for Kesch

### CODE REVIEW

**Comment by kraushm** [comment]
This only covers the first set of tests, i. e. the cookbook to check for sanity.

**Comment by kraushm** [comment]
Test will not pass in any case, since the module is not yet present in the production stack on the system.

**Comment by kraushm** [comment]
Fixes #685

File: cscs-checks/mch/fieldextra_check.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class FieldextraCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.maintainers = ['Mkr']
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]:
```suggestion
        self.maintainers = ['MKr']
```

File: cscs-checks/mch/fieldextra_check.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        self.strict_check = False
+
+
+@rfm.parameterized_test(['opt'], ['opt_omp'])
+class FieldextraAccuracy(FieldextraCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = (
+            'Fieldextra validation test (accuracy); MCH'
+        )
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: It should fit in a single line

File: cscs-checks/mch/fieldextra_check.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            'ln -s ${EBROOTFIELDEXTRA}/tools tools',
+            'ln -s ${EBROOTFIELDEXTRA}/resources resources',
+        ]
+
+        if variant == 'opt':
+            self.modules = ['fieldextra/12.7.5-gmvolf-17.02']
+            self.executable_opts = ['-c gnu -m opt']
+        else:
+            self.modules = ['fieldextra/12.7.5-gmvolf-17.02-openmp']
+            self.executable_opts = ['-c gnu -m opt_omp']
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]:
```suggestion
            self.executable_opts = ['-c', 'gnu', '-m', 'opt_omp']
```

File: cscs-checks/mch/fieldextra_check.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        self.num_tasks = 1
+        self.num_tasks_per_node = 1
+        self.num_cpus_per_task = 24
+        self.num_task_per_core = 1
+        self.use_multithreading = True
+        self.strict_check = False
+
+
+@rfm.parameterized_test(['opt'], ['opt_omp'])
+class FieldextraAccuracy(FieldextraCheck):
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]:
```suggestion
class FieldextraAccuracyTest(FieldextraCheck):
```

File: cscs-checks/mch/fieldextra_check.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class FieldextraCheck(rfm.RunOnlyRegressionTest):
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]:
```suggestion
class FieldextraTestBase(rfm.RunOnlyRegressionTest):
```

File: cscs-checks/mch/fieldextra_check.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                'OMP_STACKSIZE': '500M',
+                'OMP_NUM_THREADS': '$SLURM_CPUS_PER_TASK',
+            }
+
+        self.sanity_patterns = sn.assert_found(r'All tests successful',
+                                               self.stdout)
+
+
+@rfm.simple_test
+class FieldextraPerformance(FieldextraCheck):
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]:
```suggestion
class FieldextraPerformanceTest(FieldextraCheck):
```

File: cscs-checks/mch/fieldextra_check.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    def __init__(self):
+        super().__init__()
+        self.descr = (
+            'Fieldextra validation test (performance); MCH'
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: Single line.

File: cscs-checks/mch/fieldextra_check.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        )
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'Fieldextra', 'performance')
+        self.modules = ['fieldextra/12.7.5-gmvolf-17.02-openmp']
+        self.executable = 'fieldextra_gnu_opt_omp'
+        self.executable_opts = ['nl.TC']
+        self.readonly_files = [
+            '18112900_204',
+        ]
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: Single line.

File: cscs-checks/mch/fieldextra_check.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+        self.pre_run = [
+            'source create_nl_6h.template'
+        ]
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: Single line.

File: cscs-checks/mch/fieldextra_check.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        self.sanity_patterns = sn.assert_found(
+            r'%INFO fieldextra: Program successfully completed', self.stdout
+        )
+        self.perf_patterns = {
+            'perf': sn.extractsingle(r'WALL CLOCK\s*SPEEDUP\D*(?P<perf>\S+)',
+                                     'fieldextra.diagnostic', 'perf', float)
+        }
+        self.reference = {
+            'kesch': {
+                'perf': (350., None, 0.20)
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: Why don't you put the current value with a lower threshold? Unless we are fine with the 20%.

File: cscs-checks/mch/fieldextra_check.py [hunk 11]
```diff
@@ -22,12 +22,10 @@ def __init__(self):
...
 
 @rfm.parameterized_test(['opt'], ['opt_omp'])
-class FieldextraAccuracy(FieldextraCheck):
+class FieldextraAccuracyTest(FieldextraTestBase):
     def __init__(self, variant):
         super().__init__()
-        self.descr = (
-            'Fieldextra validation test (accuracy); MCH'
-        )
+        self.descr = ('Fieldextra validation test (accuracy); MCH')
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: No parentheses here.

File: cscs-checks/mch/fieldextra_check.py [hunk 12]
```diff
@@ -56,23 +54,17 @@ def __init__(self, variant):
...
 
 @rfm.simple_test
-class FieldextraPerformance(FieldextraCheck):
+class FieldextraPerformanceTest(FieldextraTestBase):
     def __init__(self):
         super().__init__()
-        self.descr = (
-            'Fieldextra validation test (performance); MCH'
-        )
+        self.descr = ('Fieldextra validation test (performance); MCH')
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: No parentheses here.

File: cscs-checks/mch/fieldextra_check.py (hunk 11 above)
Comment (vkarak) [reply cscs-checks/mch/fieldextra_check.py]: Also remove the `MCH` for all descriptions. You should tag the tests with `mch`.

**Pull request merged**